File: 1501-1750/1653. Minimum Deletions to Make String Balanced/solution.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        dp = 0 
        bs = 0
        for c in s : 
            if c == 'b' : 
                bs += 1 
            else :
                dp = min( dp + 1 , bs )
        return dp 

File: 1501-1750/1653. Minimum Deletions to Make String Balanced/solution.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        int n = s.size() ; 
        int dp = 0 ; 
        int bs = ( s[0] == 'b' ? 1 : 0 ) ; 
        for( int i=1 ; i<n ; ++i ){
            if( s[i] == 'b' ){
                bs++ ; 
            }else {
                dp = min( dp + 1 , bs ) ; 
            }
        }   
        return dp ; 
    }
};